cluster/gce/container-linux/configure-helper.sh

```diff
@@ -690,7 +734,12 @@ function start-kube-proxy {
   fi
   local container_env=""
   if [[ -n "${ENABLE_CACHE_MUTATION_DETECTOR:-}" ]]; then
-    container_env="env:\n    - name: KUBE_CACHE_MUTATION_DETECTOR\n    value: \"${ENABLE_CACHE_MUTATION_DETECTOR}\""
+    # Daemonset manifest needs a different indent.
+    if [ ${is_daemonset_manifest} = true ]; then
+      container_env="env:\n        - name: KUBE_CACHE_MUTATION_DETECTOR\n          value: \"${ENABLE_CACHE_MUTATION_DETECTOR}\""
+    else
+      container_env="env:\n    - name: KUBE_CACHE_MUTATION_DETECTOR\n      value: \"${ENABLE_CACHE_MUTATION_DETECTOR}\""
```

Comment:
I suggest setting an indent variable to "  " and setting container_env only once with the variable expanded in the string. The difference between these strings is difficult to pick out.

Reply:
Logic is removed. See the new commit on #50705.